q2_assembly/quast/tests/test_quast.py

```diff
@@ -791,15 +794,246 @@ def _copy(dst_dir):
                 make_artifact=make_artifact,
             )
 
-            tab_report, _ = evaluate_contigs(ctx=mock_ctx, contigs=contigs)
-            tab_report.validate()
+            refs_dir = self.get_data_path("references")
+            _copy_references(refs_dir, tmp)
 
+            refs = Artifact.import_data("GenomeData[DNASequence]", refs_dir)
+            tab_report, _, _ = evaluate_contigs(
+                ctx=mock_ctx, contigs=contigs, references=refs
+            )
             make_artifact.assert_called_once_with(
                 "QUASTResults", self.assert_is_dataframe((2, 43))
             )
+
+        tab_report.validate()
+        self.assertEqual(action.call_args[0][0], contigs)
+        export_data.assert_called_once_with(os.path.join(tmp, "vis_files"))
+
+    def test_evaluate_contigs_pipeline_no_refs_normal(self):
+        # this is used for mocking
+        def _copy(dst_dir):
+            os.makedirs(dst_dir, exist_ok=True)
+            shutil.copy2(
+                src=os.path.join(
+                    self.get_data_path("quast-results"), "enhanced_tabular_results.tsv"
+                ),
+                dst=os.path.join(dst_dir, "quast_results.tsv"),
+            )
```

Comment:
You can put this method on `self` directly - then it will become accessible in other tests, like below.